crates/sol-macro/src/expand/contract.rs

```diff
@@ -332,14 +332,29 @@ pub(super) fn expand(cx: &ExpCtxt<'_>, contract: &ItemContract) -> Result<TokenS
             )
         }));
 
+        let filter_methods = events.iter().map(|&e| {
+            let event_name = cx.overloaded_name(e.into());
+            let name = format_ident!("{name}_filter");
+            let doc = format!(
+                "Creates a new event filter for the [`{event_name}`] event.",
+            );
+            quote! {
+                #[doc = #doc]
+                pub fn #name(&self) -> alloy_contract::Event<N, T, &P, #event_name> {
+                    self.event_filter::<#event_name>()
+                }
+            }
```

Comment:
Should we make these default snake case? I realize that for `function doStuff` we codegen `foo.doStuff().send()` atm

Reply:
Currently we default everything to the same name as declared in solidity/ABI. It's possible in the future to implement renaming like in serde but currently there is no real demand.